### PR TITLE
migrate test_xml to pytest

### PR DESCRIPTION
**Pull request by frogunder**
### What does this PR do?

migrate test_xml to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/modules/test_xml.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import os
+import tempfile
+
+import pytest
+
+from salt.modules import xml
+
+
+@pytest.fixture
+def XML_STRING():
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_xml.py:15]:
```suggestion
def XML_STRING():
```
Lowercase the fixture name

File: tests/pytests/unit/modules/test_xml.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    Verify xml.get_value
+    """
+    with tempfile.NamedTemporaryFile("w+", delete=False) as xml_file:
+        xml_file.write(XML_STRING)
+        xml_file.flush()
+
+    xml_result = xml.get_value(xml_file.name, ".//actor[@id='2']")
+    assert xml_result == "Liam Neeson"
+
+    os.remove(xml_file.name)
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_xml.py:48]:
```suggestion
def test_get_value(XML_STRING, tmp_path):
    """
    Verify xml.get_value
    """
    xml_file = tmp_path / "name-it-as-you-like.xml"
    xml_file.write_text(XML_STRING)
    xml_result = xml.get_value(str(xml_file), ".//actor[@id='2']")
    assert xml_result == "Liam Neeson"
```
Do the same for the other tests

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_xml.py (hunk 2 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_xml.py]: Thank you

**Comment by frogunder** [comment]
re-run all

**Review by s0undt3ch — DISMISSED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**